### PR TITLE
Wishlist implemented.

### PR DESCRIPTION
**Pull request by RadmanHi**

### CODE REVIEW

File: src/main/java/com/ecommerce/galleon/controller/WishListController.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        User user = authenticationService.getUser(token);
+
+        // save the item in wishlist
+
+        WishList wishList = new WishList(user, product);
+
+        wishListService.createWishlist(wishList);
+
+        ApiResponse apiResponse = new ApiResponse(true, "Added to wishlist");
```
Comment (ali-moghadam) [inline src/main/java/com/ecommerce/galleon/controller/WishListController.java:47]: It's better to have a resource file for texts. may we need to handle texts in other languages in future.

File: src/main/java/com/ecommerce/galleon/controller/WishListController.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        return  new ResponseEntity<>(apiResponse, HttpStatus.CREATED);
+
+    }
+
+
+    // get all wishlist item for a user
+
+    @GetMapping("/{token}")
```
Comment (ali-moghadam) [inline src/main/java/com/ecommerce/galleon/controller/WishListController.java:55]: I think we should pass the token in the **header** instead of **path**

**Review by ali-moghadam — APPROVED** [review]

File: src/main/java/com/ecommerce/galleon/controller/WishListController.java (hunk 1 above)
Comment (RadmanHi) [reply src/main/java/com/ecommerce/galleon/controller/WishListController.java:47]: Sure I will fix it🙏

File: src/main/java/com/ecommerce/galleon/controller/WishListController.java (hunk 2 above)
Comment (RadmanHi) [reply src/main/java/com/ecommerce/galleon/controller/WishListController.java:55]: That's true, Authentication of the stable version of application will be implemented using Spring Security or oAuth library which handle the token management in an out of the box manner and these problems would be solved✅

**Pull request merged**